core/Files.cc

```diff
@@ -17,211 +18,75 @@ namespace {
 
 constexpr auto EXTERNAL_PREFIX = "external/com_stripe_ruby_typer/"sv;
 
-struct SigilInfo {
-    LocOffsets loc;
-    StrictLevel strictLevel;
-};
-
-SigilInfo strictSigilInfo(string_view source) {
-    /*
-     * StrictLevel::None: <none>
-     * StrictLevel::False: # typed: false
-     * StrictLevel::True: # typed: true
-     * StrictLevel::Strict: # typed: strict
-     * StrictLevel::String: # typed: strong
-     * StrictLevel::Autogenerated: # typed: autogenerated
-     */
-    size_t start = 0;
-    while (true) {
-        start = source.find("typed:", start);
-        if (start == string_view::npos) {
-            return {LocOffsets::none(), StrictLevel::None};
-        }
+template <class Sigil> class ParseSigil {
+public:
+    static Sigil parse(string_view source) {
+        size_t start = 0;
+        while (true) {
+            start = source.find(SigilTraits<Sigil>::SIGIL_PREFIX, start);
+            if (start == string_view::npos) {
+                return SigilTraits<Sigil>::NONE;
+            }
 
-        auto comment_start = start;
-        while (comment_start > 0) {
-            --comment_start;
-            auto c = source[comment_start];
-            if (c == ' ') {
+            auto comment_start = start;
+            while (comment_start > 0) {
+                --comment_start;
+                auto c = source[comment_start];
+                if (c == ' ') {
+                    continue;
+                } else {
+                    break;
+                }
+            }
+            if (source[comment_start] != '#') {
+                ++start;
                 continue;
-            } else {
-                break;
             }
-        }
-        if (source[comment_start] != '#') {
-            ++start;
-            continue;
-        }
 
-        start += 6;
-        while (start < source.size() && source[start] == ' ') {
-            ++start;
-        }
+            start += 6;
```

Comment:
```suggestion
            start += SigilTraits<Sigil>::PREFIX.size();
```

I think this is more correct.